src/optimizer/statistics/operator/propagate_filter.cpp

```diff
@@ -242,6 +242,7 @@ unique_ptr<NodeStatistics> StatisticsPropagator::PropagateStatistics(LogicalFilt
 			i--;
 			if (filter.expressions.empty()) {
 				// just break. The physical filter planner will plan a projection instead
+				// we don't remove the filter because it might have a projection map.
```

Comment:
I know you've only added a comment here, but we can check whether `LogicalFilter::projection_map` is empty rather than assuming it's not